### PR TITLE
fix(core): fix transient select error after symbol column rename sequence

### PR DESCRIPTION
**Pull request by ideoma**
Fix TableReader reload error after a series column renames found by fuzz tests.

The symptom is the error:

```
Caused by: io.questdb.griffin.SqlException: [0] [2]: could not open read-only [file=/var/folders/kj/wpw2cl295rbdksb3ysqsj1yw0000gn/T/junit12416700228706723530/dbRoot/db/testFuzzWithSnapshot_2~7/new_col_0.k]
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlException.position(SqlException.java:103)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlOptimiser.openReaderAndEnumerateColumns(SqlOptimiser.java:2523)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlOptimiser.enumerateTableColumns(SqlOptimiser.java:1772)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlOptimiser.enumerateTableColumns(SqlOptimiser.java:1777)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlOptimiser.optimise(SqlOptimiser.java:4705)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlCompilerImpl.compileExecutionModel0(SqlCompilerImpl.java:1336)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlCompilerImpl.compileExecutionModel(SqlCompilerImpl.java:1325)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlCompilerImpl.compileUsingModel(SqlCompilerImpl.java:1418)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlCompilerImpl.compileInner(SqlCompilerImpl.java:1389)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.griffin.SqlCompilerImpl.compile(SqlCompilerImpl.java:231)
	at io.questdb@7.3.3-SNAPSHOT/io.questdb.cairo.pool.SqlCompilerPool$C.compile(SqlCompilerPool.java:98)
	at com.questdb/com.questdb.cairo.wal.transfer.ReplicationFuzzTest.assertReplicatedTable(ReplicationFuzzTest.java:271)
	at com.questdb/com.questdb.cairo.wal.transfer.ReplicationFuzzTest.lambda$testFuzzWithSnapshot$7(ReplicationFuzzTest.java:553)
```

The fix is to check that the symbol column name has the same `nameTxn` suffix as the one already opened by ``SymbolMapReaderImpl`

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 19 (94.74%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|10|11|90.91%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/SymbolMapReaderImpl.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cairo/TableReader.java [hunk 1]
```diff
@@ -648,8 +648,16 @@ private void copyColumns(
     }
 
     private void copyOrRenewSymbolMapReader(SymbolMapReader reader, int columnIndex) {
-        if (reader != null && reader.isDeleted()) {
-            reader = reloadSymbolMapReader(columnIndex, reader);
+        if (reader != null) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/TableReader.java:651]: is there a way to siplify the code?

`copyOrRenewSymbolMapReader` lacks `copy` action. It reloads symbol conditionally using `reloadSymbolMapReader`.

Both `copyOrRenewSymbolMapReader` and `reloadSymbolMapReader` are called only from `reshuffleSymbolMapReaders` and they overlap on functionality

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/cairo/TableReader.java (hunk 1 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/TableReader.java]: ok, refactored in [6ade81c](https://github.com/questdb/questdb/pull/3787/commits/6ade81cc57825cfd169d6e53340fa52d6ae4e15e)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 25 / 27 (92.59%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|17|19|89.47%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/SymbolMapReaderImpl.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**